src/main/java/ProjectOcean/Controller/YearController.java

```diff
@@ -77,10 +91,38 @@ private void onDragRelease(DragEvent event) {
         model.addCourse(icon.getICourse(), year.getID(), studyPeriod, slot);
 
         event.setDropCompleted(true);
-        moveDraggedObjectToCursor.moveDraggedObjectToCursor(icon, event);
+        relocateDraggedObjectToCursor.relocateDraggedObjectToCursor(icon, event);
         event.consume();
     }
 
+    /**
+     * Deletes the this from the observer list in the model.
+     */
+    public void deleteFromObserver(){
+        model.deleteObserver(this);
+    }
+
+    public void setGreenBorderColorInSlots(String studyPeriod){
+        //TODO gör sen
+        System.out.println("YEARGRID CHildren:" + yearGrid.getChildren());
+
+        List<Node> slots = yearGrid.getChildren();
+
+        for (Node slot: slots) {
+            //Make sure that the group that displays the gridlines are not counted.
+            if(slots.indexOf(slot) != 0) {
+               // System.out.println("SLOT: " + slot);
+                Integer column = 1 + GridPane.getColumnIndex(slot);
```

Comment:
Why Integer instead of int?

Reply:
Because GridPane.getColumnIndex returns it as a reference variable. So this is to match the output given.